### PR TITLE
chore(backend): setup `dancer2` backend for `theweeklychallenge`

### PR DESCRIPTION
**Pull request by baimamboukar**
This Pull Request closes #126 in https://github.com/manwar/TheWeeklyChallenge/issues/126) by introducing the complete setup for the `Dancer2` backend for routes in https://github.com/manwar/TheWeeklyChallenge/issues/125.

- Bootstrapped `Dancer2` backend project.
- Implemented `MVC` pattern structure.
- Added `controllers` for Public routes.
- Introduced 'Team' and 'User' models.
- Integrated `templates` for Public routes.
- Imported `controllers` modules in the main file.
- Configured `Dockerfile` for backend `containerization`.
- Added a `Makefile` for convenient command execution.
- Implemented a `.gitignore` file to untrack unnecessary files.

These changes collectively enhance the Dancer2 backend, following MVC best practices and enabling containerization for efficient deployment.

### CODE REVIEW

**Comment by baimamboukar** [comment]
Hello @manwar 👋

Here is a PR draft, I will be updating it as I'm progressing and promote it  as `ready for review` when I'm done

**Comment by manwar** [comment]
@baimamboukar I have a suggestion, keep the root namespace PWC::
I know the project is "The Weekly Challenge" then why "PWC"? It is because we started as "Perl Weekly Challenge". Even today we call the team "Team PWC"
Group code in the MVC style i.e. folder structure like Models/, Views/ and Controllers/

Also please get in touch with @himanshisrestha as she is working on landing page. You would need this to begin with. Rest can be added later.

**Comment by baimamboukar** [comment]
@manwar 

- This is noted, and I will rename the project 🫡
- The project has been initialized using the `Dancer2 CLI`, so the `views` folder is created automatically. I create the `Models` and `Controllers` directories inside the `lib` directory. Does this work for you, @manwar?

**Comment by manwar** [comment]
@baimamboukar "views" folder is the place for template if I am not wrong. The "Views" folder is backend code responsible for view part of MVC.

**Comment by baimamboukar** [comment]
> @baimamboukar "views" folder is the place for template if I am not wrong. The "Views" folder is backend code responsible for view part of MVC.

Yes @manwar. That's right 

So we will have:
- `views` for Templates (created by default with Dancer2 CLI)
- Then the `lib` directory will contain the `MVC` pattern setup
Like..

```markdown
lib
 |_Models
 |_Views
 |_Controllers
```
Is it all good?

**Comment by baimamboukar** [comment]
Hello @manwar 👋
This is ready for review

**Comment by manwar** [comment]
@baimamboukar There is some confusion and I must have misled you.

When I suggested PWC as root namespace I meant this:
  
      lib/Models/PWC/YourPackage.pm
      lib/Views/PWC/YourPackage.pm
      lib/Controllers/PWC/YourPackage.pm

Also when you name a package don't include word "Controller" or 'View" or "Model", the full namespace of the package makes it clear e.g. Controller::PWC::YourPackage

**Comment by baimamboukar** [comment]
> @baimamboukar There is some confusion and I must have misled you.
> 
> When I suggested PWC as root namespace I meant this:
> 
> ```
>   lib/Models/PWC/YourPackage.pm
>   lib/Views/PWC/YourPackage.pm
>   lib/Controllers/PWC/YourPackage.pm
> ```
> 
> Also when you name a package don't include word "Controller" or 'View" or "Model", the full namespace of the package makes it clear e.g. Controller::PWC::YourPackage

Noted @manwar ! I'm on it

File: src/lib/Controllers/PWC/AboutController.pm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package AboutController;
```
Comment (manwar) [inline src/lib/Controllers/PWC/AboutController.pm]: Please remove the word "Controller" from the package name.
Also the package doesn't make sense?
If I am not wrong it should have been like below:

      package Controllers::PWC::About;

Comment (baimamboukar) [reply src/lib/Controllers/PWC/AboutController.pm]: Thanks for the quick review @manwar ❤️‍🔥

This is noted.


Should I also rename the file ? Like, from `AboutController.pm` to `About.pm` ?

Comment (manwar) [reply src/lib/Controllers/PWC/AboutController.pm]: Yes, it has be to same as the package name.

Comment (baimamboukar) [reply src/lib/Controllers/PWC/AboutController.pm]: Done @manwar!

**Comment by manwar** [comment]
Good job, thanks @baimamboukar

**Comment by manwar** [comment]
@baimamboukar I noticed you have everything under the "src/". I have moved everything out to the parent folder.
I noticed you are not in the Slack. I have created separate workspace just for this project. You can join us there.
Please share your email and I can send you invitation.

**Comment by baimamboukar** [comment]
> @baimamboukar I noticed you have everything under the "src/". I have moved everything out to the parent folder. I noticed you are not in the Slack. I have created separate workspace just for this project. You can join us there. Please share your email and I can send you invitation.

Thank you for your feedback, @manwar . I appreciate your input.

**Comment by baimamboukar** [comment]
> @baimamboukar I noticed you have everything under the "src/". I have moved everything out to the parent folder. I noticed you are not in the Slack. I have created separate workspace just for this project. You can join us there. Please share your email and I can send you invitation.

Oooh! Sorry... The link was broken when I tried to join....
You can send the invitation to `baimamboukar@gmail.com`

Thanks @manwar 🫡✨

**Comment by manwar** [comment]
@baimamboukar sent the invitation.

**Pull request merged**